### PR TITLE
logaggregator: do not send backlog when requested lines are zero

### LINKED ISSUE #2120: logaggregator: Full backlog is returned when lines=0
Full backlog is returned when `lines=0`. The reason being `backlog` is being set [here](https://github.com/flynn/flynn/blob/master/logaggregator/api.go#L69) without checking that `lines > 0`.

### PR DESCRIPTION
**Pull request by imjorge**
Closes #2120

### CODE REVIEW

File: logaggregator/api_test.go [hunk 1]
```diff
@@ -172,6 +172,70 @@ func (s *LogAggregatorTestSuite) TestAPIGetLogFollow(c *C) {
...
+		select {
+		case l := <-lines:
+			if l.err != nil {
+				return "", fmt.Errorf("could not read log output: %s", l.err)
+			}
+			return l.text, nil
+		case <-time.After(1 * time.Second):
+			return "", errors.New("timed out waiting for log output")
+		}
+	}
```
Comment (titanous) [inline logaggregator/api_test.go]: Looks like this re-uses all of the setup from `TestAPIGetLogFollow`, should we abstract into a setup method?

Comment (imjorge) [inline logaggregator/api_test.go]: We may. I will do it.

--jmf

**Comment by imjorge** [comment]
Merged tests `TestAPIGetLogFollow` and `TestAPIGetLogFollowWithoutBacklog`.

**Comment by titanous** [comment]
LGTM, thanks!

**Pull request merged**